cocos/2d/components/ui-opacity.ts

```diff
@@ -133,8 +134,32 @@ export class UIOpacity extends Component {
     @serializable
     protected _opacity = 255;
 
+    protected _parentAdded (node: Node): void {
+        const parent: Node |null = node.getParent();
+        if (parent == null) {
+            return;
+        }
+        const render = parent._uiProps.uiComp as UIRenderer;
+        if (render && render.color) {
+            return;
+        }
+        const uiOp = parent.getComponent<UIOpacity>(UIOpacity);
+        if (uiOp != null) {
+            UIOpacity.setEntityLocalOpacityDirtyRecursively(node, true, uiOp.opacity / 255, true);
+        }
+    }
+
+    protected _parentRemoved (node: Node): void {
+        const uiOp = node.getComponent<UIOpacity>(UIOpacity);
+        if (uiOp) {
+            UIOpacity.setEntityLocalOpacityDirtyRecursively(node, true, 1, false);
+        }
+    }
+
     public onEnable (): void {
-        // If the ancestor node has a uiopacity component, it will be initialized when initializing 
+        this.node.on(NodeEventType.CHILD_ADDED, this._parentAdded, this);
+        this.node.on(NodeEventType.CHILD_REMOVED, this._parentRemoved, this);
```

Comment:
Need to turn off the listener in the onDisable callback

Reply:
OK